### PR TITLE
Add `qmk find` command, reuse logic for `qmk mass-compile`.

### PR DESCRIPTION
**Pull request by tzarc**
## Description

As per the title.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: lib/python/qmk/search.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    cli.log.setLevel(level)
+    logging.root.setLevel(level)
+    cli.release_lock()
+    return old
+
+
+def _all_keymaps(keyboard):
+    old = _set_log_level(logging.CRITICAL)
+    keymaps = qmk.keymap.list_keymaps(keyboard)
+    _set_log_level(old)
```
Comment (zvecr) [inline lib/python/qmk/search.py:28]: Could add something like,
```python
import contextlib

@contextlib.contextmanager
def ignore_logging():
    old = _set_log_level(logging.CRITICAL)
    yield
    _set_log_level(old)
```

then these functions can become a little safer and do....

```py
def _all_keymaps(keyboard):
    with ignore_logging():
        keymaps = qmk.keymap.list_keymaps(keyboard)
```

Comment (tzarc) [reply lib/python/qmk/search.py:28]: TIL. Let me have a play.

File: lib/python/qmk/search.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            for filter_txt in filters:
+                f = equals_re.match(filter_txt)
+                if f is not None:
+                    key = f.group('key')
+                    value = f.group('value')
+                    cli.log.info(f'Filtering on condition ("{key}" == "{value}")...')
+
+                    def _make_filter(k, v):
+                        expr = fnmatch.translate(v)
+                        rule = re.compile(expr, re.IGNORECASE)
```
Comment (zvecr) [inline lib/python/qmk/search.py:80]: Does the change from `re.compile(f'^{expr}$', re.IGNORECASE)` change any behaviour?

Comment (tzarc) [reply lib/python/qmk/search.py:80]: That's actually unintentional. I think I must've copy-pasted it from the old `master` branch.

**Review by zvecr — APPROVED** [review]

**Pull request merged**